pennylane/math/single_dispatch.py

```diff
@@ -645,6 +646,19 @@ def _sum_torch(tensor, axis=None, keepdims=False, dtype=None):
 ar.register_function("torch", "cond", _cond)
 
 
+def _matmul_torch(tensor1, tensor2, out=None):

```

Comment:
So I think this is actually the case where we should be using `multidispatch`.  The multi-dispatch dispatch figures out the best interface for a multi-interface situation. So if both numpy and torch are ever used in conjunction, it uses torch to perform operations.

Reply:
I believe this is already done by autoray. If you do:
```pycon
>>> t1 = torch.tensor([1])
>>> t2 = [1]
>>> type(qml.math.matmul(t2, t1))
torch.Tensor
```
and viceversa.

My understanding was that we should use `multi_dispatch` when we want to define logic for multiple interfaces, and `single_dispatch` when defining the logic for a single interface.

Reply:
@AlbertMitjans oh that must be new 🤔 Autoray used to only ever dispatch on the first argument of all functions

Reply:
@josh146 you're right! In the example above `np.matmul` is called, but `numpy` casts the result to a torch Tensor. I will add it to `multi_dispatch` then.

Reply:
Done @albi3ro